src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_user/block.rs

```diff
@@ -0,0 +1,784 @@
+            "vhost-user-blk: get features done, ret:{:?}, queue_size: {:?}",
+            avail_features,
+            queue_sizes.len()
+        );
+
+        // for the standard vhost_user_blk device, get the device config from slave.
+        let config_space = {
+            master.set_features(avail_features)?;
+            let protocol_featuers = master.get_protocol_features()?;
+            // set the config features to get the device's config from slave.
+            master.set_protocol_features(protocol_featuers)?;
+
+            let config_len = mem::size_of::<VirtioBlockConfig>();
+            let config_space: Vec<u8> = vec![0u8; config_len as usize];
+
+            let (_, mut config_space) = master
+                .get_config(
+                    VHOST_USER_CONFIG_OFFSET,
+                    config_len as u32,
+                    VhostUserConfigFlags::WRITABLE,
+                    config_space.as_slice(),
+                )
+                .map_err(VirtIoError::VhostError)?;
+
+            // set the num queues
+            config_space[CONFIG_SPACE_SIZE - 2] = init_queues as u8;
+            config_space[CONFIG_SPACE_SIZE - 1] = (init_queues >> 8) as u8;
+            config_space
+        };
+
+        let mut intr_evts = Vec::with_capacity(init_queues as usize);
+        for _ in 0..init_queues {
+            intr_evts.push(EventFd::new(0).map_err(VirtIoError::IOError)?);
+        }
+
+        Ok(VhostUserBlockDevice {
+            vhost_socket,
+            queue_sizes: queue_sizes.clone(),
+            device_info: VirtioDeviceInfo::new(
+                VHOST_USER_BLOCK_DRIVER_NAME.to_string(),
+                avail_features,
+                queue_sizes,
+                config_space,
+                event_mgr,
+            ),
+            endpoint: Endpoint::new(
+                master,
+                MASTER_SLOT,
+                VHOST_USER_BLOCK_DRIVER_NAME.to_string(),
+            ),
+            timer_fd: Some(TimerFd::new().map_err(VirtIoError::IOError)?),
+            intr_evts: Arc::new(intr_evts),
+            curr_queues: init_queues,
+            id: VHOST_USER_BLOCK_DRIVER_NAME.to_string(),
+        })
+    }
+
+    fn reconnect_to_server(&mut self) -> VirtIoResult<Master> {
+        if !Path::new(self.vhost_socket.as_str()).exists() {
+            return Err(VirtIoError::InternalError);
+        }
+        let master = Master::connect(&self.vhost_socket, 1).map_err(VirtIoError::VhostError)?;
+
+        Ok(master)
+    }
+
+    // vhost-user protocol features this device supports
+    fn get_dev_protocol_features(&self) -> VhostUserProtocolFeatures {
+        let mut features = VhostUserProtocolFeatures::MQ;
+
+        // TODO: need to support INFLIGHT_SHMFD later.
+        features |= VhostUserProtocolFeatures::CONFIG
+            | VhostUserProtocolFeatures::CONFIGURE_MEM_SLOTS
+            | VhostUserProtocolFeatures::REPLY_ACK;
+        // | VhostUserProtocolFeatures::INFLIGHT_SHMFD;
```

Comment:
Since the feature is not supported yet, it is better to delete the line instead of commenting it out. Just TODO is fine!

Reply:
OK, I will add a TODO here.